src/lib.rs

```diff
@@ -320,6 +320,224 @@ impl DltExtendedHeader {
+            0x1 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogFatal)),
+            0x2 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogError)),
+            0x3 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogWarn)),
+            0x4 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogInfo)),
+            0x5 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogDebug)),
+            0x6 => Some(Mtin::DltLogMessage(MtinDltLogMessage::DltLogVerbose)),
+            //Reserved values
+            _ => None,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MtinDltTraceMessage {
+    ///Value of variable.
+    DltTraceVariable = 0x1,
+    ///Call of a function.
+    DltTraceFunctionIn = 0x2,
+    ///Return of a function.
+    DltTraceFunctionOut = 0x3,
+    ///State of a state machine.
+    DltTraceState = 0x4,
+    ///RTE Events.
+    DltTraceVfb = 0x5,
+}
+
+impl MtinDltTraceMessage {
+    ///Attempts to convert a raw mtin value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mtin> {
+        match value {
+            0x1 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceVariable)),
+            0x2 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceFunctionIn)),
+            0x3 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceFunctionOut)),
+            0x4 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceState)),
+            0x5 => Some(Mtin::DltTraceMessage(MtinDltTraceMessage::DltTraceVfb)),
+            //Reserved values
+            _ => None,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MtinDltNetworkMessage {
+    ///Inter-Process-Communication.
+    DltNwTraceIpc,
+    ///CAN communication bus.
+    DltNwTraceCan,
+    ///FlexRay communication bus.
+    DltNwTraceFlexray,
+    ///Most communication bus.
+    DltNwTraceMost,
+    ///Ethernet communication bus.
+    DltNwTraceEthernet,
+    ///SOME/IP communication.
+    DltNwTraceSomeIp,
+    ///User defined settings.
+    DltNwUserDefined(u8),
+}
+
+impl MtinDltNetworkMessage {
+    ///Attempts to convert a raw mtin value to the enum.
+    ///Returns `Option::None` for reserved values.
+    fn from_u8(value: u8) -> Option<Mtin> {
+        match value {
+            0x1 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceIpc)),
+            0x2 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceCan)),
+            0x3 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceFlexray)),
+            0x4 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceMost)),
+            0x5 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceEthernet)),
+            0x6 => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwTraceSomeIp)),
+            //User defined
+            0x7 ..= 0xF => Some(Mtin::DltNetworkMessage(MtinDltNetworkMessage::DltNwUserDefined(value))),
```

Comment:
reserved values actually go up to 0x15 in the spec but since the mtin is only 4 bit I assume 0xF was meant here

Reply:
Makes sense. Probably a forgotten conversion when they switched from decimal to hex numbers.